pinot-core/src/main/java/org/apache/pinot/core/segment/processing/mapper/SegmentMapper.java

```diff
@@ -129,10 +133,11 @@ public Map<String, GenericRowFileManager> map()
   private Map<String, GenericRowFileManager> doMap()
       throws Exception {
     Consumer<Object> observer = _processorConfig.getProgressObserver();
-    int totalCount = _recordReaderFileConfigs.size();
     int count = 1;
+    int totalCount = _recordReaderFileConfigs.size();
     GenericRow reuse = new GenericRow();
-    for (RecordReaderFileConfig recordReaderFileConfig : _recordReaderFileConfigs) {
+    for (int i = 0; i < _recordReaderFileConfigs.size(); i++) {
+      RecordReaderFileConfig recordReaderFileConfig = _recordReaderFileConfigs.get(i);
       RecordReader recordReader = recordReaderFileConfig._recordReader;
```

Comment:
Can we create a public static method in RecordReaderFileConfig that takes care of creating recordReader and assigning it to the recordReader instance (for that particular RecordReaderFileConfig). Basically we can keep all the allocation, clean up contained within RecordReaderFileConfig, for readability.

Reply:
I have done that in the current iteration. Having RecordReaderFileConfig handling recordreader related work(allocation, clean up, checks etc.) makes sense.